kratos/includes/kratos_application.h

```diff
@@ -101,6 +105,8 @@ class KRATOS_API(KRATOS_CORE) KratosApplication
     void RegisterMATVariables(); //TODO: move to application
     void RegisterLegacyStructuralAppVariables(); //TODO: move to application
 
+    std::string Name(){return mApplicationName;}
```

Comment:
I would make this method const. Also it would be better to return the const&. I understand that you are using move semantic but I would avoid it as we have already allocated object.